File: ImaginaryWebshop.Web/src/app/models/product.model.ts
export interface ProductCreateDto {
    name: string;
    description: string;
    price: number;
    category: string;
    pictureUrl: string;
}

export interface ProductUpdateDto {
    name?: string;
    description?: string;
    price?: number;
    category?: string;
    pictureUrl?: string;
}

export interface ProductDetailsDto {
    id: string;
    name: string;
    description: string;
    price: number;
    category: string;
    pictureUrl: string;
}

File: ImaginaryWebshop.Data/Models/ProductModel.cs
namespace ImaginaryWebshop.Data.Models
{
    public class ProductModel
    {
        [Key] public Guid Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public double Price { get; set; }
        public string Category { get; set; } = string.Empty;
        public string? PictureUrl { get; set; }
    }
}
